ports/rp2/machine_uart.c

```diff
@@ -246,8 +373,10 @@ STATIC mp_uint_t machine_uart_read(mp_obj_t self_in, void *buf_in, mp_uint_t siz
                 }
             }
             MICROPY_EVENT_POLL_HOOK
+            // force few incoming bytes to the buffer
+            uart_drain_rx_fifo(self, true);
```

Comment:
Is this needed?  I would think that the UART IRQ would trigger and update the ringbuf if there were any characters (and then locking is not needed).

Reply:
The irq is triggered on transiting a certain level in the FIFO, which can be configured is set to 4 during init as the minimal level. Less than 4 characters will not trigger the IRQ.